relayer/src/services/relayer/optimism.ts

```diff
@@ -76,6 +76,34 @@ export class OptimismRelayerService extends BaseRelayerService {
     return;
   }
 
+  async safeRelayMessage(message: CrossChainMessage): Promise<boolean> {
+    try {
+      // Finalize message via OptimismPortal (it calls the L1CrossDomainMessenger internally)
+      (await this.messenger.finalizeMessage(message)).wait(1);
+    } catch (err) {
+      if (err.message.includes("OptimismPortal: withdrawal has already been finalized")) {
+        // Here, the case was that OptimismPortal marked the withdrawal as finalized but the relay still failed
+        // We need to call the L1CrossDomainMessenger to relay the message for usinstead (not going via the OptimismPortal)
+        console.log("Try relaying via messenger..");
+        const crossChainMessage = await this.messenger.toCrossChainMessage(message);
+        const withdrawal = await this.messenger.toLowLevelMessage(crossChainMessage);
+
+        const tx = await this.l1Wallet.sendTransaction({
+          to: this.messenger.contracts.l1.L1CrossDomainMessenger.address,
+          // withdrawal.message contains the complete transaction data
+          data: withdrawal.message,
+        });
+
+        await tx.wait(1);
+      } else if (err.message.includes("CrossDomainMessenger: message has already been relayed")) {
```

Comment:
Is it `CrossDomainMessenger` or `L1CrossDomainMessenger`?

Reply:
Its CrossDomainMessenger. This is how the exception is defined in their contracts.

Reply:
![Screenshot 2023-06-09 at 13 05 05](https://github.com/beamer-bridge/beamer/assets/18466333/29b5019c-60fc-4424-a586-303378a4ae38)